exporters/inmemory/src/main/java/io/opentelemetry/exporters/inmemory/InMemoryMetricExporter.java

```diff
@@ -11,11 +27,68 @@
 
 /**
  * A {@link MetricExporter} implementation that can be used to test OpenTelemetry integration.
+ *
+ * <p>Can be created using {@code InMemoryMetricExporter.create()}
+ *
+ * <p>Example usage:
+ *
+ * <pre><code>
+ * public class InMemoryMetricExporterExample {
+ *
+ *   // creating InMemoryMetricExporter
+ *   private final InMemoryMetricExporter exporter = InMemoryMetricExporter.create();
+ *   private final MeterSdkProvider meterSdkProvider = OpenTelemetrySdk.getMeterProvider();
+ *   private final Meter meter = meterSdkProvider.get("InMemoryMetricExporterExample");
+ *   private IntervalMetricReader intervalMetricReader;
+ *
+ *   void setup() {
+ *     intervalMetricReader =
+ *         IntervalMetricReader.builder()
+ *             .setMetricExporter(exporter)
+ *             .setMetricProducers(Collections.singletonList(meterSdkProvider.getMetricProducer()))
+ *             .setExportIntervalMillis(1000)
+ *             .build();
+ *   }
+ *
+ *   List&lt;MetricData&gt; readExportedMetrics() {
+ *     return exporter.getFinishedMetricItems();
+ *   }
+ *
+ *   void shutdown() {
+ *     intervalMetricReader.shutdown();
+ *   }
+ *
+ *   LongCounter generateLongMeter(String name) {
+ *     return meter.longCounterBuilder(name).setDescription("Sample LongCounter").build();
+ *   }
+ *
+ *   DoubleCounter generateDoubleMeter(String name) {
+ *     return meter.doubleCounterBuilder(name).setDescription("Sample DoubleCounter").build();
+ *   }
+ *
+ *
+ *   public static void main(String[] args) throws InterruptedException {
+ *     InMemoryMetricExporterExample example = new InMemoryMetricExporterExample();
+ *
+ *     example.setup();
+ *     example.generateLongMeter("counter-1");
```

Comment:
I would just inline all methods, but `setup()` and use a single meter. The idea is to have just a minimal example to showcase the usage.